storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteTopicPartitionDirectory.java

```diff
@@ -0,0 +1,169 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.log.remote.storage;
+
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.internals.Topic;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Objects.requireNonNull;
+import static java.util.regex.Pattern.compile;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.RemoteLogSegmentFileType.getUuid;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.deleteFilesOnly;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.deleteQuietly;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Represents a topic-partition directory in the local tiered storage under which filesets for
+ * log segments are stored.
+ *
+ *
+ * <code>
+ * / storage-directory / uuidBase64-0-topic / tvHCaSDsQZWsjr5rbtCjxA.log
+ *                     .                   .  tvHCaSDsQZWsjr5rbtCjxA.index
+ *                     .                   .  tvHCaSDsQZWsjr5rbtCjxA.timeindex
+ *                     .
+ *                     / 5fEBmixCR5-dMntYSLIr1g-3-topic / BFyXlC8ySMm-Uzxw5lZSMg.log
+ *                                                      . BFyXlC8ySMm-Uzxw5lZSMg.index
+ *                                                      . BFyXlC8ySMm-Uzxw5lZSMg.timeindex
+ * </code>
+ */
+public final class RemoteTopicPartitionDirectory {
+    private static final Logger LOGGER = getLogger(RemoteTopicPartitionDirectory.class);
+    private static final String UUID_LEGAL_CHARS = "[a-zA-Z0-9_-]{22}";
+
+    /**
+     * The format of a Kafka topic-partition directory. Follows the structure Uuid-partition-Topic.
+     */
+    private static final Pattern FILENAME_FORMAT = compile("(" + UUID_LEGAL_CHARS + ")-(\\d+)-(" + Topic.LEGAL_CHARS + "+)");
+    private static final int GROUP_UUID = 1;
+    private static final int GROUP_PARTITION = 2;
+    private static final int GROUP_TOPIC = 3;
```

Comment:
In case we want to adopt the new structure:
```suggestion
    private static final Pattern FILENAME_FORMAT = compile("(" + Topic.LEGAL_CHARS + "+)-(\\d+)-(" + UUID_LEGAL_CHARS + ")");
    private static final int GROUP_UUID = 3;
    private static final int GROUP_PARTITION = 2;
    private static final int GROUP_TOPIC = 1;
```